frontend/src/metabase/dashboard/context/context.tsx

```diff
@@ -121,69 +125,84 @@ const DashboardContextProviderInner = ({
   navigateToNewCardFromDashboard,
   ...reduxProps
 }: PropsWithChildren<ContextProps>) => {
-  const [error, setError] = useState<FailedFetchDashboardResult | null>(null);
+  const dispatch = useDispatch();
+  const [error, setError] = useState<unknown | null>(null);
   const previousIsLoading = usePrevious(isLoading);
   const previousIsLoadingWithoutCards = usePrevious(isLoadingWithoutCards);
 
+  const [dashboardId, setDashboardId] = useState<DashboardId | null>(null);
   const previousDashboard = usePrevious(dashboard);
   const previousDashboardId = usePrevious(dashboardId);
   const previousTabId = usePrevious(selectedTabId);
   const previousParameterValues = usePrevious(parameterValues);
 
   const shouldRenderAsNightMode = Boolean(isNightMode && isFullscreen);
 
-  const handleLoadDashboard = useCallback(
-    async (dashboardId: DashboardId) => {
-      initialize({ clearCache: !isNavigatingBackToDashboard });
-
-      const result = await fetchDashboard({
-        dashId: dashboardId,
-        queryParams: parameterQueryParams,
-        options: {
-          clearCache: !isNavigatingBackToDashboard,
-          preserveParameters: isNavigatingBackToDashboard,
-        },
-      });
-
-      return result;
-    },
-    [
-      fetchDashboard,
-      initialize,
-      isNavigatingBackToDashboard,
-      parameterQueryParams,
-    ],
-  );
-
   const handleError = useCallback(
-    (err: FailedFetchDashboardResult | Error) => {
-      const error = err instanceof Error ? { error: err, payload: err } : err;
+    (error: unknown) => {
       onError?.(error);
       setError(error);
     },
     [onError],
   );
 
-  useEffect(() => {
-    const hasDashboardChanged = dashboardId !== previousDashboardId;
-    if (hasDashboardChanged) {
-      setError(null);
-      handleLoadDashboard(dashboardId)
-        .then((result) => {
-          if (isFailedFetchDashboardResult(result)) {
-            handleError(result);
-          }
-        })
-        .catch((err) => {
-          handleError(err);
-        });
-      return;
-    }
+  const fetchId = useCallback(
+    async (idToFetch: DashboardId) => {
+      try {
+        const { id, isError } = await dispatch(
+          fetchEntityId({ type: "dashboard", id: idToFetch }),
```

Comment:
I may be missing some context here, why do we need to use `entity_ids` at all? As far as I'm aware of not all entities have them, there are many dashboards without entity_ids: https://stats.metabase.com/question/26647-dashboards-without-entity-ids